src/DynamoUtilities/CLIWrapper.cs

```diff
@@ -145,7 +145,7 @@ protected virtual async Task<string> GetData(int timeoutms)
                     return writer.ToString();
                 }
             });
-            var completedTask = await Task.WhenAny(readStdOutTask, Task.Delay(TimeSpan.FromMilliseconds(timeoutms)));
+            var completedTask = Task.WhenAny(readStdOutTask, Task.Delay(TimeSpan.FromMilliseconds(timeoutms))).Result;
             //if the completed task was our read std out task, then return the data
             //else we timed out, so return an empty string.
             return completedTask == readStdOutTask ? readStdOutTask.Result : string.Empty;
```

Comment:
we might be able to simplify lines 148 to 151 by using
```
return readStdOutTask.Wait(timeoutms) ? readStdOutTask.Result : string.Empty;
```
https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.wait?view=net-8.0#system-threading-tasks-task-wait(system-int32)

Reply:
either way we go, please test that the timeout can be reached without issues.
Ex Add a Thread.Sleep(timeoutms *2) in the TaskRun delegate.

Reply:
@pinzart90 - does this test already cover what you are concerned about -or not?
https://github.com/DynamoDS/Dynamo/blob/master/test/Libraries/DynamoUtilitiesTests/CLIWrapperTests.cs#L47

Reply:
@pinzart90 I've added another test and slightly modified the GetData method to make it possible to inject a mock readLine delegate that will be invoked inside the Task delegate so we can sleep or return data etc.